packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback, useContext } from 'react';
+import { useFeatureIsOn } from '../components/GrowthBookProvider';
+import { FeedPostClick } from './feed/useFeedOnPostClick';
+import { Post } from '../graphql/posts';
+import { Features } from '../lib/featureManagement';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+
+import { useExtensionPermission } from './useExtensionPermission';
+import { useLazyModal } from './useLazyModal';
+import { LazyModal } from '../components/modals/common/types';
+import { useActions } from './useActions';
+import { ActionType } from '../graphql/actions';
+
+type CompanionTriggerProps = {
+  post: Post;
+  index?: number;
+  row?: number;
+  column?: number;
+};
+
+type OpenArticle = (data?: CompanionTriggerProps) => () => Promise<void>;
+
+export type CompanionTrigger = {
+  onFeedArticleClick: FeedPostClick;
+  onPostArticleClick: PostClick;
+  activateCompanion: () => void;
+  openArticle: OpenArticle;
+  toggleOpen: (value: boolean, data?: CompanionTriggerProps) => void;
+};
+
+type PostClick = (e: React.MouseEvent, data: { post: Post }) => Promise<void>;
+
+export default function useCompanionTrigger(
+  customPostClickHandler: (
+    post: Post,
+    index?: number,
+    row?: number,
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { closeModal, openModal } = useLazyModal();
+  const { isActionsFetched, checkHasCompleted, completeAction } = useActions();
+
+  const featureEnabled = useFeatureIsOn(
+    Features.EngagementLoopJuly2023Companion,
+  );
```

Comment:
This is a essential part, we have to ensure this will only trigger when you are on the extension + on a main feed page.
Any other case should not even trigger it, it has to do with when the experiment needs to start for users.

Personally think we can basically have a early exit if you are not on extension + mainFeed *(there is some example of the later bit for onboarding redirection for example)

Reply:
Gotcha, that makes sense, nice catch! Since this is not just checking the flag, but also enrolling the user in the experiment. I'll update 👍